CrudApp.Tests/Repositories/BookRepositoryTests.cs

```diff
@@ -0,0 +1,289 @@
+            Title = "Test Title",
+            Author = "Test Author",
+            PublishYear = 1900,
+        };
+
+        var bookJSON = fileDbConnection.ConvertBookToJSON(newBook);
+
+        var sut = testHelper
+            .SetupAddBook(bookJSON)
+            .CreateSut();
+
+        // Act
+        var response = sut.AddBook(newBook);
+
+        // Assert
+        response.Should().Be((true, null)!);
+    }
+
+    [Fact]
+    public void CanAddBook_WithCorrectJSON()
+    {
+        // Setup
+        var testHelper = new TestHelper();
+        var fileService = new FileService();
+        var filePath = "mockpath";
+        var fileDbConnection = new FileDbConnection(new FileService(), filePath);
+        var newBook = new Book()
+        {
+            Id = 1,
+            Title = "Test Title",
+            Author = "Test Author",
+            PublishYear = 1900,
+        };
+
+        var bookJSON = fileDbConnection.ConvertBookToJSON(newBook);
+
+        var sut = testHelper
+            .SetupAddBook(bookJSON)
+            .CreateSut();
+
+        // Act
+        var response = sut.AddBook(newBook);
+
+        // Assert
+        testHelper._bookPostAdding.Should().Be(bookJSON);
+    }
+
+    [Fact]
+    public void CanRemoveBook_WithSuccess()
+    {
+        // Setup
+        var testHelper = new TestHelper();
+        var book1 = new Book()
+        {
+            Id = 1,
+            Title = "Book 1",
+            Author = "Author 1",
+            PublishYear = 1901,
+        };
+
+        var book2 = new Book()
+        {
+            Id = 2,
+            Title = "Book 2",
+            Author = "Author 2",
+            PublishYear = 1902,
+        };
+
+        var books = new List<Book>
+        {   book1,
+            book2
+        };
+
+        var booksPreRemoval = books.ConvertAll(book => JsonConvert.SerializeObject(book));
+        var booksPostRemoval = booksPreRemoval
+            .Skip(1)
+            .ToList();
+
+        var sut = testHelper
+            .SetupRemoveBook(booksPreRemoval, booksPostRemoval)
+            .CreateSut();
+
+        // Act
+        sut.RemoveBook(book1.Title);
+
+        // Assert
+        testHelper._booksPostRemoval.Should().BeEquivalentTo(booksPostRemoval);
+    }
+
+    [Fact]
+    public void CanUpdateBook_WithSuccess()
+    {
+        // Setup
+        var testHelper = new TestHelper();
+        var fileService = new FileService();
+        var filePath = "mockpath";
+        var fileDbConnection = new FileDbConnection(new FileService(), filePath);
+        var originalBook = new Book()
+            {
+                Id = 1,
+                Title = "Book 1",
+                Author = "Author 1",
+                PublishYear = 1901,
+            };
+
+        var updatedBook = new Book()
+        {
+            Id = 2,
+            Title = "Book 2",
+            Author = "Author 2",
+            PublishYear = 1902,
+        };
+
+        var originalBookJSON = new List<string>()
```

Comment:
Yeah - no. Sorry. A `List<Book>` should serialize to `[{},{}]` rather than a `List<string>`